### PR TITLE
Fix Corgium

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

Fixes various issues related to corgium.
* Players no longer become catatonic after dying and ghosting
* Corgium no longer equips non-dog fashion hats
* Dead players are no longer stuck as corgis

## Why It's Good For The Game

Bug fixes are fairly radical.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>
Hats without dog fashion won't equip. 

image
image

Ghosted players no-longer go catatonic.
image
image
image

</details>

## Changelog
:cl:
Fix: Fix corgium catatonic bug
Fix: Fix corgium hat bug
Tweak: Assign default values to shapeshift
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
See https://github.com/BeeStation/BeeStation-Hornet/pull/4128 It has a pretty good solution

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 1]
```diff
@@ -517,6 +517,12 @@
 							/datum/reagent/napalm, /datum/reagent/hellwater, /datum/reagent/toxin/plasma, /datum/reagent/toxin/spore_burning)
 	var/active = 0
 
+/obj/item/reagent_containers/food/drinks/bottle/molotov/corgium
+	name = "molotov cocktail"
+	desc = "A throwing weapon used to ignite things, typically filled with an accelerant. Recommended highly by rioters and revolutionaries. Light and toss."
+	icon_state = "vodkabottle"
```
Comment (PowerfulBacon) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm]: Doesn't need this if its a subtype

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 2]
```diff
@@ -517,6 +517,12 @@
 							/datum/reagent/napalm, /datum/reagent/hellwater, /datum/reagent/toxin/plasma, /datum/reagent/toxin/spore_burning)
 	var/active = 0
 
+/obj/item/reagent_containers/food/drinks/bottle/molotov/corgium
```
Comment (PowerfulBacon) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm]: Why is this a molotov? Won't it just not work at all as a molotov?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -125,47 +125,37 @@
...
 			new_corgi.place_on_head(hat,null,FALSE)
-	L.forceMove(new_corgi)
+	H = new(new_corgi,src,L)
 
 /datum/reagent/corgium/on_mob_life(mob/living/carbon/M)
 	. = ..()
 	//If our corgi died :(
 	if(new_corgi.stat)
 		holder.remove_all_type(type)
+		addtimer(CALLBACK(src, .proc/restore, M), 2 SECONDS)
```
Comment (PowerfulBacon) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:147]: Does on_mob_life get called on the death of the corgi even?

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm (hunk 1 above)
Comment (DrDuckedGoose) [reply code/modules/food_and_drinks/drinks/drinks/bottle.dm]: leftover debugging, apologies

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 3 above)
Comment (DrDuckedGoose) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:147]: The reagent is still inside and processing inside the original mob. This means every ``on_mob_life()`` the stat of the corgi is checked. So when the corgi dies, the original mob is still processing the reagent, that checks the if the corgi is dead, if it is it deletes the corgi, moving the original mob out, and applies the corgi's damage to the mob.
image

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**